pkg/api/jokes.go

```diff
@@ -35,30 +34,41 @@ func (h Handler) addJoke(w http.ResponseWriter, r *http.Request) {
 	title := r.FormValue("title")
 	body := r.FormValue("body")
 
-	h.storage.AddJoke(title, body)
+	err := h.storage.AddJoke(title, body)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+
+		_, err := w.Write([]byte(err.Error()))
+		if err != nil {
+			log.Printf("Response writing error: %s", err)
+		}
+
+		return
+	}
 
 	http.Redirect(w, r, "/jokes", http.StatusFound)
 }
 
-func (h Handler) getJoke(w http.ResponseWriter, r *http.Request) {
+func (h Handler) getJokesByText(w http.ResponseWriter, r *http.Request) {
 	text := r.URL.Query().Get("text")
-	id := r.URL.Query().Get("id")
-	skip, seed, err := h.GetPaginationParams(r)
+
+	skip, seed, err := getPaginationParams(r)
 	if err != nil {
 		w.WriteHeader(http.StatusBadRequest)
 	}
 
 	if text != "" {
```

Comment:
you can simplify the whole function if you rewrite it with
`if text == "" {
w.WriteHeader(http.StatusBadRequest)
return 
}
`